homeassistant/util/color.py

```diff
@@ -417,14 +417,59 @@ def color_rgb_to_rgbw(r: int, g: int, b: int) -> tuple[int, int, int, int]:
 
 def color_rgbw_to_rgb(r: int, g: int, b: int, w: int) -> tuple[int, int, int]:
     """Convert an rgbw color to an rgb representation."""
-    # Add the white channel back into the rgb channels.
+    # Add the white channel to the rgb channels.
     rgb = (r + w, g + w, b + w)
 
     # Match the output maximum value to the input. This ensures the
     # output doesn't overflow.
     return _match_max_scale((r, g, b, w), rgb)  # type: ignore
 
 
+def color_rgb_to_rgbww(
+    r: int, g: int, b: int, min_mireds: int, max_mireds: int
+) -> tuple[int, int, int, int, int]:
+    """Convert an rgb color to an rgbww representation."""
+    # Find the color temperature when both white channels have equal brightness
+    mired_range = max_mireds - min_mireds
+    mired_midpoint = min_mireds + mired_range / 2
+    color_temp_kelvin = color_temperature_mired_to_kelvin(mired_midpoint)
+    w_r, w_g, w_b = color_temperature_to_rgb(color_temp_kelvin)
+
+    # Find the ratio of the midpoint white in the input rgb channels
+    white_level = min(r / w_r, g / w_g, b / w_b)
+
+    # Subtract the white portion from the rgb channels.
+    rgb = (r - w_r * white_level, g - w_g * white_level, b - w_b * white_level)
+    rgbww = (*rgb, round(white_level * 255), round(white_level * 255))
```

Comment:
This will enable both white channels. Maybe both channels should be set to half brightness instead

Reply:
So, making it half, makes sense in a way. However, it makes RGB(255,255,255) weird?

Reply:
Right, it's weird either way, and the light may or may not compensate for brightness

Reply:
Let's start with this... As in, its a good start 👍